proto/regen/data/v1alpha2/types.proto

```diff
@@ -0,0 +1,209 @@
+syntax = "proto3";
+
+package regen.data.v1alpha2;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+option go_package = "github.com/regen-network/regen-ledger/x/data";
+
+// ContentID specifies a hash based content identifier for a piece of data
+message ContentHash {
+    oneof descriptor {
+        Raw raw = 1;
+        Graph graph = 2;
+        Geo geo = 3;
+    }
+
+    message Raw {
+        // hash represents the hash of the data based on the specified digest_algorithm
+        bytes hash = 1;
+
+        // digest_algorithm represents the hash digest algorithm.
+        DigestAlgorithm digest_algorithm = 2;
+
+        // media_type represents the MediaType for raw data.
+        MediaType media_type = 3;
+
+    }
+
+    message Graph {
+        // hash represents the hash of the data based on the specified digest_algorithm
+        bytes hash = 1;
+
+        // digest_algorithm represents the hash digest algorithm.
+        DigestAlgorithm digest_algorithm = 2;
+
+        // graph_canonicalization_algorithm represents the RDF graph canonicalization algorithm. It should
+        // be left unset if type is not ID_TYPE_GRAPH.
+        GraphCanonicalizationAlgorithm canonicalization_algorithm = 3;
+
+        GraphMerkleTree merkle_tree = 4;
```

Comment:
What does this value actually tell us about the hash? Does the hash then represent the merkle tree root? I think i'm a little confused as to what different values of this enum actually mean in practice for how we handle a given Graph's content hash.

Reply:
Added some comment suggestions to clarify this.